### PR TITLE
RTK + Query with Example

### PR DESCRIPTION
**Pull request by AndreaLandiArk**
Template project base

- RTK
- RTKQ
- React 18
- Node 20
- Chackra UI

Added example apiSlice with data fetch from a public test API

### CODE REVIEW

**Review by AndreaCuneo — CHANGES_REQUESTED** [review]

File: src/components/sideBar/menuItem/menuItem.tsx [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+    <ChakraLink
+      _hover={{ textDecoration: "none" }}
+      as={ReactRouterLink}
+      to={path}
+      isExternal={isExternal}
+    >
+      <WrapItem
+        _hover={{
+          background: "brand.primary",
+          color: "white",
```
[the remark this answers is not in the document]
Comment (AndreaLandiArk) [reply src/components/sideBar/menuItem/menuItem.tsx:28]: i set this in the theme but it is still not working. i comment there

File: src/theme.tsx [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+      dark: "#194069",
+      darken: "#282c37",
+    },
+  },
+  fonts: {
+    heading: `'PTSansRegular', sans-serif`,
+    body: `'PTSansRegular', sans-serif`,
+  },
+  components: {
+    WrapItem: {
```
Comment (AndreaLandiArk) [inline src/theme.tsx:27]: @AndreaCuneo  I set all the style here but it seems to be ignoring it

File: src/components/header/view.tsx [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+
+const Header = () => {
+  return (
+    <Box
+      as="header"
+      position="fixed"
+      top="0"
+      left="0"
+      right="0"
+      zIndex="999" // You can adjust the z-index as needed
```
Comment (AndreaCuneo) [inline src/components/header/view.tsx:21]: why not instructing to use chakra zIndex named?

File: src/components/helper.ts [hunk 4]
```diff
@@ -0,0 +1,6 @@
+export function generateUniqueKey(
```
Comment (AndreaCuneo) [inline src/components/helper.ts:1]: this is dangerous as generate a new key on every call. if this is called within a Component, the Children would get rerendered every time the parent is. 
Are you sure?

File: src/redux/modules/errorHandler.ts [hunk 5]
```diff
@@ -0,0 +1,162 @@
...
+        title: requestFailedTitle,
+        message: err.errorMessage,
+        originalTitle: err?.response?.title,
+        originalDetail: err?.response?.detail,
+        displayStatus: true,
+      },
+    });
+  };
+
+export const Selectors: { all: (a: any) => errorModalType } = {
```
Comment (AndreaCuneo) [inline src/redux/modules/errorHandler.ts:160]: any doesn't look right.

File: src/theme.tsx (hunk 2 above)
Comment (AndreaCuneo) [reply src/theme.tsx:27]: no idea, check Chakra docs.
But is still "white", should not be brand.text / brand.textAlt ?

**Review by AndreaCuneo — CHANGES_REQUESTED** [review]

File: src/components/helper.ts (hunk 4 above)
Comment (AndreaLandiArk) [reply src/components/helper.ts:1]: added this for a test, forgot to delete

**Review by AndreaCuneo — APPROVED** [review]

**Pull request merged**